Source/Extensions/Blazorise.Snackbar/Blazorise.Snackbar.csproj

```diff
@@ -13,6 +13,8 @@
 
   <ItemGroup>
     <ProjectReference Include="..\..\Blazorise\Blazorise.csproj" />
+    <ProjectReference Include="..\..\SourceGenerators\Blazorise.ApiDocsGenerator\Blazorise.ApiDocsGenerator.csproj" OutputItemType="Analyzer" ReferenceOutputAssembly="false" PrivateAssets="all" />
```

Comment:
I feel like adding ApiDocsGenerator to every project will bite us in the ***. I remember we had problems(and still have) with `Blazorise.Generator.Features.csproj`. Even though it is marked as `PrivateAssets="all,"`, we still had to publish it as a separate Nuget package because it wouldn't work without it.

Reply:
I hit exactly this issue yesterday when trying to figure out the `[ShouldBeOnlyUseInternally]` attribute. Creating such an attribute in a project referenced with `PrivateAssets="all"` and using it in the "parent" project causes problems. While `PrivateAssets="all"` ensures that the `SourceGeneratorProject.dll` isn't included in the NuGet package, it doesn't prevent the code from `SourceGeneratorProject` from being used in the source code. This ultimately results in the NuGet package not functioning as expected.

In this case, the `ApiDocsGenerator` project (like `ApiDocsGeneratorDtos`) isn't used in the projects from which we are trying to obtain documentation. For instance, there is no `using ApiDocsGenerator;` directive in the `Blazorise` or `Blazorise.AnExtension` projects. This contrasts with `Blazorise.Generator.Features`, which behaves differently. Therefore, I think it’s safe to proceed in this scenario.

By the way, I can think of a solution where using the attribute wouldn't require packing the source generator project either.

Reply:
Can we use Fody to remove any traces of the code that is needed for the generator?

Reply:
What about Fody? Since we're merging into 1.7 we should make sure that we don't have to release additional NuGet package. We need to strip everything not needed for Blazorise to work in production.